Sources/PackageLoading/Module+PkgConfig.swift

```diff
@@ -133,29 +133,31 @@ extension SystemPackageProvider {
 /// compiler/linker. List of allowed flags:
 /// cFlags: -I, -F
 /// libs: -L, -l, -F, -framework
-func whitelist(pcFile: String, flags: (cFlags: [String], libs: [String])) throws {
+func whitelist(pcFile: String, flags: (cFlags: [String], libs: [String])) -> (cFlags: [String], libs: [String])  {
     // Returns an array of flags which doesn't match any filter.
     func filter(flags: [String], filters: [String]) -> [String] {
         var filtered = [String]()     
         var it = flags.makeIterator()
         while let flag = it.next() {
-            guard let filter = filters.filter({ flag.hasPrefix($0) }).first else {
+            if let filter = filters.filter({ flag.hasPrefix($0) }).first {
                 filtered += [flag]
-                continue
-            }
-            // If the flag and its value are separated, skip next flag.
-            if flag == filter {
-                guard let _ = it.next() else {
-                   fatalError("Expected associated value") 
+
+                // If the flag and its value are separated, skip next flag.
+                if flag == filter {
+                    guard let next = it.next() else {
+                        fatalError("Expected associated value")
+                    }
+                    filtered += [next]
                 }
             }
+
         }
         return filtered
     }
-    let filtered = filter(flags: flags.cFlags, filters: ["-I", "-F"]) + filter(flags: flags.libs, filters: ["-L", "-l", "-F", "-framework"])
-    guard filtered.isEmpty else {
-        throw PkgConfigError.nonWhitelistedFlags("Non whitelisted flags found: \(filtered) in pc file \(pcFile)")
-    }
+    return (
```

Comment:
Instead of silently discarding all flags, we think each flag should be handled on case by case basis. So, we should discard these two flags: `-fabi-version -fno-omit-frame-pointer` but still fail for other unknown flags.